### PR TITLE
bazel: update bazel-lib

### PR DESCRIPTION
**Pull request by keith**
Fixes https://github.com/envoyproxy/envoy/issues/22217

Signed-off-by: Keith Smiley <keithbsmiley@gmail.com>

### CODE REVIEW

File: bazel/repository_locations.bzl [hunk 1]
```diff
@@ -119,8 +119,8 @@ REPOSITORY_LOCATIONS_SPEC = dict(
         project_name = "Aspect Bazel helpers",
         project_desc = "Base Starlark libraries and basic Bazel rules which are useful for constructing rulesets and BUILD files",
         project_url = "https://github.com/aspect-build/bazel-lib",
-        version = "1.3.1",
-        sha256 = "4ef2f746bae7bd7f1ec39dc9b53a9d7e8002f18233ea2c2ee4702bbb5283c7ca",
+        version = "1.7.0",
+        sha256 = "58fab45b4419a5eaa340489d567a013e80334c8e4bd998288149cd7e39437586",
```
Comment (phlax) [inline bazel/repository_locations.bzl:123]: you need to adjust the date but otherwise lgtm

Comment (keith) [reply bazel/repository_locations.bzl:123]: ah crap always gets me

Comment (keith) [reply bazel/repository_locations.bzl:123]: thanks done

Comment (phlax) [reply bazel/repository_locations.bzl:123]: for ref this should work

```console

$ export GITHUB_TOKEN=$(my.secrets)
$ bazel run //tools/dependency:check -- -c release_dates

```

Comment (phlax) [reply bazel/repository_locations.bzl:123]: the secret doesnt need any power - just gh access

**Review by phlax — APPROVED** [review]
/lgtm deps

lets hope it resolves the issue - do we need to do anything else to make it work ?

**Comment by keith** [comment]
doesn't seem like it from my local testing on a m1, them removing that constraint upstream was the fix since now it can find that toolchain

**Pull request merged**